test/test.ts

```diff
@@ -1,35 +1,178 @@
 import chai from "chai";
 import chaiAsPromised from "chai-as-promised";
-import { getParsedData } from "backend-class";
-import fsCases from "./cases-fs";
-import axiosCases from "./cases-axios";
+import {
+  DataServer,
+  getBodyElement,
+  getParsedHref,
+  getParsedURL,
+  parseTree
+} from "backend-class";
+import casesDataParserURL from "./cases-data-parser/cases-url";
+import casesDataParserHREF from "./cases-data-parser/cases-href";
+import casesDataParserHTTP from "./cases-data-parser/cases-axios";
+import casesDataParserFS from "./cases-data-parser/cases-fs";
+import casesDataServerHTTP from "./cases-data-server/cases-axios";
+import casesDataServerFS from "./cases-data-server/cases-fs";
 // Note: baseUrl is set to src/ https://stackoverflow.com/a/43721170
 
 // Note: test async with chai-as-promised https://stackoverflow.com/a/45496509
 chai.use(chaiAsPromised);
 
 describe("Data parser test", () => {
+  // Test URL parsers
+  describe("Test URL parser", () => {
+    // Positive cases
+    casesDataParserURL.forEach(([desc, url, href, file]) => {
```

Comment:
I've actually not seen test cases setup before, where the it() description is also imported. I like it though